osu.Game.Tests/Gameplay/TestSceneStoryboardSamples.cs

```diff
@@ -70,6 +73,37 @@ public void TestSamplePlaybackAtZero()
             AddUntilStep("sample playback succeeded", () => sample.LifetimeEnd < double.MaxValue);
         }
 
+        [Test]
+        public void TestSamplePlaybackWithRateMods()
+        {
+            GameplayClockContainer gameplayContainer = null;
+            TestDrawableStoryboardSample sample = null;
+
+            OsuModDoubleTime doubleTimeMod = null;
+
+            AddStep("create container", () =>
+            {
+                var beatmap = Beatmap.Value = CreateWorkingBeatmap(new OsuRuleset().RulesetInfo);
+
+                Add(gameplayContainer = new GameplayClockContainer(beatmap, new[] { doubleTimeMod = new OsuModDoubleTime() }, 0));
+
+                SelectedMods.Value = new[] { doubleTimeMod };
+                Beatmap.Value = new TestCustomSkinWorkingBeatmap(beatmap.Beatmap, gameplayContainer.GameplayClock, Audio);
```

Comment:
The ordering is *very* incorrect, the beatmap should be created first along the mods, then the container will be created with providing what has been set (i.e.):

**Though this depends on another suggestion in the construction parameters of `TestCustomSkinWorkingBeatmap`.**

```suggestion
                Beatmap.Value = new TestCustomSkinWorkingBeatmap(new OsuRuleset().RulesetInfo, Audio);
                SelectedMods.Value = new[] { doubleTimeMod };
                
                Add(gameplayContainer = new GameplayClockContainer(Beatmap.Value, SelectedMods.Value, Audio);
```